### DISCUSSION TITLE
Videos take a long time to start

### DISCUSSION THREAD

**Discussion by xorcat** [ressu/kube-plex#36] [Q&A]
Hey there, thanks for bringing this project into the present day!

I've got it mostly working in my home lab: https://github.com/xorcat/infra/blob/main/cluster/apps/media/helm-release.yaml

This pulls from your HelmRepo, and it's largely working, except when starting to play a video, it can take a minute or so before playback begins (seems to be proportional to file size). Once playback has begun I'm able to scrub around and pause almost instantly. This doesn't seem to happen when playing original either. 

Seems like this is something to do with the beginning of the transcoding process, although transcode pods are brought up really quickly, and storage appears to attach fine. 

Do you have any ideas on where I can start to troubleshoot this?

If it matters, I'm playing on an AppleTV 4K HDR, but I've also tried on my phone and iPad - wherever it's trying to transcode I seem to have this problem. 

All of the volumes are the same StorageClass (media-fs), which is a CephFS volume. 

Thanks again!
Aaron

**Comment by xorcat** [comment]
Alright, I don't know the answer yet, but I'm pretty confident its my storage, and not [kube-]plex.

For some reason my cephfs storage needs to buffer some of/the whole file in the CSI driver before it starts presenting it to the container, for some reason; which means I'm waiting for the file to get to some level of availability before it starts being sent to the transcode container.

I'm trying to dig further into this, but I'm pretty sure there's nothing related to kube-plex here :)

**Comment by ressu** [comment]
I would have suspected either pod scheduling or storage, but as you mentioned there is a storage delay. I'm thinking that it might be a good idea to add reporting to the kube-plex binary that periodically checks the state of the transcoding job and if the job isn't up and running returns a status message with the status of the job.

That should give more visibility to how transcoding is performing internally. Even if it wouldn't have really given too much information in this case.

**Reply by xorcat** [reply]
I'm going on a journey with ceph on this - at this stage I _think_ it's expected behaviour and my files are just big/network slow. I also think there's a debug flag to change the behaviour in ceph. 

I'll try to remember to come back with my findings in case anybody else is interested.
